lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -22,6 +24,7 @@ def self.run(params)
         fad_api_client = Client::FirebaseAppDistributionApiClient.new
         platform = Actions.lane_context[Actions::SharedValues::PLATFORM_NAME]
         binary_path = params[:ipa_path] || params[:apk_path]
+        fad_api_client.set_auth_token(params[:google_service_account])
```

Comment:
+1 yeah interested in your thinking behind using a setter versus passing it to the constructor 🙂

Reply:
Thats a good point. I originally had it in the constructor and I moved it once I moved the auth client out. Since it only needed to be passed through I didn't think it needed a variable since we did not care about that path persisting. I have changed the way it gets passed to make it less confusing overall. Now it gets stored as a variable in the constructor and the variable will be passed in whenever auth_token needs to exist. This is in contrast to calling the set method to set this auth token. Thanks for the comment.